### PR TITLE
Move util.StringSet into its own package

### PR DESCRIPTION
**Pull request by tummychow**
This is part of #13698 (see also #3486 and #4851). Most of it was done using `gofmt -r` and sed. This contribution is under the corporate CLA signed by Square Inc.

Summary of changes:
- move util/set.go and util/set_test.go into their own package, util/stringset
- rename util.NewStringSet to stringset.New
- update references to the moved identifiers
- update imports

### CODE REVIEW

**Comment by tummychow** [comment]
Sorry about that, some out-of-order imports needed to be corrected with gofmt. I wasn't sure if this project would be comfortable with goimports (which did some aggressive import reordering).

**Comment by bgrant0607** [comment]
cc @lavalamp @krousey

**Comment by brendandburns** [comment]
LGTM, but needs rebase.

**Comment by brendandburns** [comment]
@k8s-bot ok to test

**Comment by tummychow** [comment]
rebased onto master

**Comment by lavalamp** [comment]
Is it too late to bikeshed about the package name? `stringset.StringSet` is very, very clunky. Ideally we'd end up with `set.String` (implying the possibility of auto-generating other useful sets) or `stringset.Type` (ugly) or `strings.Set` or `containers.StringSet`. Maybe that last is my favorite.

Not the end of the world if not.

**Comment by tummychow** [comment]
I have no opinions on the name of the package or type; I'd be glad to update the PR accordingly if maintainers agree on a more desirable name.

**Comment by brendandburns** [comment]
I'd bet OK with pkg/util/sets/String which would make the type sets.String

@lavalamp let's get this settled today though so we can get this PR submitted and not make busy work for developers

**Comment by lavalamp** [comment]
OK. Let's go with pkg/util/sets and type name String, final answer. @tummychow thanks for bearing with us! @k8s-oncall please merge this quickly after it's adjusted and rebased.

**Comment by tummychow** [comment]
Rebased and renamed.

File: pkg/controller/resourcequota/resource_quota_controller_test.go [hunk 1]
```diff
@@ -103,11 +103,11 @@ func TestFilterQuotaPods(t *testing.T) {
 			Status: api.PodStatus{Phase: api.PodFailed},
 		},
 	}
-	expectedResults := util.NewStringSet("pod-running",
+	expectedResults := sets.NewString("pod-running",
```
Comment (tummychow) [inline pkg/controller/resourcequota/resource_quota_controller_test.go]: It looks like this line caused the CI failure for `verify-flags-underscore.py`. Should I add it to `known-flags.txt` like the error message says?

Comment (mikedanese) [inline pkg/controller/resourcequota/resource_quota_controller_test.go]: I think it's safe to add... cc @eparis

Comment (brendandburns) [inline pkg/controller/resourcequota/resource_quota_controller_test.go]: yes, safe to add...

**Comment by tummychow** [comment]
Added `pod-running` to known-flags.txt as suggested above.

**Comment by lavalamp** [comment]
This is awesome, I've been wanting this for ages. Consider this an LGTM when everything is working.

File: pkg/controller/resourcequota/resource_quota_controller_test.go (hunk 1 above)
Comment (eparis) [inline pkg/controller/resourcequota/resource_quota_controller_test.go]: @tummychow I will update that regex in a different PR. Good idea.

**Comment by krousey** [comment]
@brendandburns merge bot bug?

**Comment by eparis** [comment]
@krousey haha, yes. seems that wa. hopefuly I'll take a few minutes with the bot tonight

**Comment by lavalamp** [comment]
@brendandburns @eparis the bug is that the merge queue gets a list and sits on it. It should re-get the list after every merge. Or at least re-get each PR before acting on it.

**Pull request merged**